library/Backends/llvm/src/Juvix/Backends/LLVM/Pass/Types.hs

```diff
@@ -164,4 +164,17 @@ data TermLLVM
   | AppM
       (Annotated TermLLVM)
       [Annotated TermLLVM]
+  | ScopedRecordDeclM RecordDecl (Annotated TermLLVM)
+  | FieldM TermFieldSelector
+  | RecordM TermRecordConstructor
   deriving (Show)
+
+type FieldDecl = (FieldName, TypeLLVM)
+
+type RecordSpec = [FieldDecl]
+
+type RecordDecl = (RecordName, RecordSpec)
+
+type TermFieldSelector = (RecordName, FieldName, Annotated TermLLVM)
+
+type TermRecordConstructor = (RecordName, [Annotated TermLLVM])
```

Comment:
Should we move the new closure code parts to where this is? It seems to me all the extensions should live close together just separated by what they extend. As IIRC the closure extensions were at the top

Reply:
I moved them in https://github.com/anoma/juvix/pull/1106/commits/02cdfb59de1f10f3c369581458d0cbaa4c5bfe39 -- does that seem like a more sensible place?